feat: Implement Emotions and Others UI components for Affective Annotation Project

teddy-f-47 opened the pull request:
They are not connected to BE yet; currently, the data are only put in JS variables. I just think that the input components may be re-usable in other modes, so I would like to make them available with this PR first.

Demo (before styling update):
https://user-images.githubusercontent.com/64476430/193839468-68734b9c-31c7-4f9b-85b2-a4ec0c35737c.mp4

Screenshots (after styling update):
<img width="960" alt="ss1" src="https://user-images.githubusercontent.com/64476430/194376446-4bfd75f6-550b-4655-b38c-c3d186045dc6.png">
<img width="960" alt="ss2" src="https://user-images.githubusercontent.com/64476430/194376460-9e9af1e2-a042-4b5c-9b33-693d911c36cc.png">
<img width="960" alt="ss3" src="https://user-images.githubusercontent.com/64476430/194376463-17f6bcc4-928b-4f96-8321-99b9e543e738.png">

What's new:
 - frontend/components/tasks/affectiveAnnotation/inputs/Slider.vue : A custom slider component that is based on v-slider. It also has a checkbox that can be enabled by setting withCheckbox to true.
 - frontend/components/tasks/affectiveAnnotation/emotions/EmotionsInput.vue : The input component for emotions data in affective annotation.
 - frontend/components/tasks/affectiveAnnotation/others/OthersInput.vue : The input component for others data in affective annotation.

What's changed:
 - frontend/i18n/* : Added translations for affective annotation emotions mode and others mode, adjusted affective annotation summary mode strings to be put inside dict.
 - frontend/components/tasks/affectiveAnnotation/inputs/TextfieldWithSeq2Seq.vue : Adjusted the code to optionally have a checkbox.
 - frontend/components/tasks/affectiveAnnotation/summary/SummaryInput.vue : Adjusted translations fields names.
 - frontend/pages/projects/_id/article-annotation/index.vue : Removed affective annotation inputs.
 - frontend/pages/projects/_id/affective-annotation/index.vue : Added affective annotation inputs.

teddy-f-47 commented:
Edit:

In the last commit, I have added the textfield input for the question "Czego życzę autorowi tego tekstu?" in Others mode, I kind of forgotten about it because I was too focusing on the slider input. 😓 

What's changed:
 - frontend/components/tasks/affectiveAnnotation/inputs/TextfieldWithSeq2Seq.vue : Adjusted the code to optionally display the checkbox.

artikandri commented on frontend/components/tasks/affectiveAnnotation/inputs/Slider.vue:
```diff
@@ -0,0 +1,144 @@
+      type: Number,
+      default: 0
+    },
+    mustClick: {
```
I wonder, is there any occurence in which the `mustClick` value can be false? Because I checked and I found no such occurence, but I might have missed something XD

teddy-f-47 replied:
Thank you very much for the feedback! I still consider myself as a newbie with vue and nuxt stuff, so it means a lot to me to know I did it okay-ish. XD

> Additionally, I need to ask: is the behavior to reset the slider to 0 after clicking and reclicking nie wiem intentional?

Yes, it is intentional. It was related to the idea that when "nie wiem" is checked, then the value for the associated category should be set to -1 (like in our discussions before, we use -1 as a form of 'null' to distinguish it from true 0 and false). However, when the "nie wiem" checkbox is unchecked, the slider will go to -1 because the data was set to -1, but there is no -1 in the slider as the min value is 0, so it goes to 0.

I think it is possible to restore the previous value before the checkbox is checked, by putting it an additional variable, but it will add some more complexity and I just don't think it is worth the effort. XD

> I wonder, is there any occurence in which the mustClick value can be false? Because I checked and I found no such occurence, but I might have missed something XD

So far I think all requirements related to affective annotation make it obligatory to click on every slider (or either on the slider or the "nie wiem" checkbox). So, the answer is no, at least so far `mustClick` value in every usage of the slider component should always be set to true. XD

I made the `mustClick` prop just in case if in the future we want to make some of the sliders optional, in which case the slider will have the thumb appear right from the beginning with a default value of 0. 😅

artikandri replied:
Alright, I understand XD 

Additionally, will this component be adjusted in the future? As for now, I think the development is very leaned towards Polish language, which is not bad if we were to compare it to the requirements, but it will not be consistent to the existing codebase   XD

teddy-f-47 replied:
> Additionally, will this component be adjusted in the future? As for now, I think the development is very leaned towards Polish language, which is not bad if we were to compare it to the requirements, but it will not be consistent to the existing codebase XD

Yes, I will move all those polish words into the i18n directory (and add the translations) and adjust the code accordingly. I think I will do that in the next PR while moving these components from article annotation page to affective annotation page XD

teddy-f-47 replied:
or maybe I will do that in this PR, but I still need some more time 😅

artikandri replied:
I think it will be better to do that in this PR, and the deadline is still far so I think it will be fine XD      I will be waiting for more updates then.

artikandri reviewed (commented):
Hello! I have checked the code and generally it looks good; good job on that! I like how you wrote the `slider` component to make it usable in other components; however I have some minor comments to check. 

Additionally, I need to ask: is the behavior to reset the slider to 0 after clicking and reclicking `nie wiem` intentional?  I know how it works in practice ofc and why did it happen, but I would need your comment on this   XD     This is just a question ofc.

teddy-f-47 commented:
I have added the translations and refactored the code a bit, in addition I also have moved all affective annotation components into the right annotation page.

teddy-f-47 commented:
Last commits adjust the styling to always show ticks on sliders, and to change the appearance of sliders to "disabled" when the "nie wiem" checkbox is marked. Similarly, also to change the appearance of textfield to "disabled" when the "nie wiem" checkbox is marked.

haanh764 reviewed (commented):
Hello, I have checked the code and it seems fine XD Thank you for such a good work. I have tested the UI and it seems oke also. I have only few small comments:
- The "Others" annotation mode, for the last question "Czego życzę autorowi tego tekstu?", I understand that if I check on the checkbox "Nie mam życzeń", then I can't add any text on the textfield. I see that you made it becomes the dot line when I clicked on the checkbox "Nie mam...", but somehow I still can add the text on the textfield, as below
![image](https://user-images.githubusercontent.com/17596118/194577492-f4d815a5-dccc-43a2-ac62-95044d34f54f.png)

- 2nd thing is pretty minor, but I think that maybe it's better to add the min and max value of the slider, something like 0-------------------------------10 (?), then the user will know how many levels they can evaluate their emotion, instead of counting the level on slider XD
- The final thing is could we somehow re-divide the proportion of height for the annotation questions part vs the text part(?) Because look at this photo, we can see that under the text there is still plenty of space while the space for annotating questions are quite less. Could we do somehow to make the height of space for text part is being fit with the height of text, then the entire space will be for annotating question XD 
![image](https://user-images.githubusercontent.com/17596118/194578582-f8d582b3-2858-4643-bf9d-2c8a13164807.png)
It's just my opinion/ question, not like a request for changing XD

haanh764 commented:
Btw, it may not be in the scope of this PR :sweat_smile: but I just recognized that when we start annotating, we start from the 1st sentence of the article, and it seems that we don't let the annotator to annotate the article title. But last time when I do the data exportation, I thought that we all agree on annotation the article title also (?) This is my question only for you or Argi since it's not really in the scope of this PR XD
![image](https://user-images.githubusercontent.com/17596118/194580867-af326b4e-db28-4f7d-a7df-c526b6f56ef6.png)

teddy-f-47 commented:
@haanh764 

1. Oh, thank you for pointing it out! I missed this, sorry about that, I will fix it soon.
2. I was thinking about that, but I vaguely remember in the meeting that that they said they don't want to show the min-max labels (?). But adding them is pretty easy so it is not a problem, maybe we can just add it and see if they'll like it too XD
3. Yeah, I see the problem, it is the behavior from the original annotation box, apparently it has some min height set, I will see if I can adjust it.

4. Yes, we also asked Piotr about this and he said the title technically should be annotated too. But I think the current code should already work for that (?) -- in your imported file, do you have the text with the type "title"?

teddy-f-47 commented:
Because if the title is only in metadata, we cant annotate it 😅 we need to have an element in the array in the imported file, which contains the text and the type: "title"

haanh764 commented:
> @haanh764
> 
> 1. Oh, thank you for pointing it out! I missed this, sorry about that, I will fix it soon.
> 2. I was thinking about that, but I vaguely remember in the meeting that that they said they don't want to show the min-max labels (?). But adding them is pretty easy so it is not a problem, maybe we can just add it and see if they'll like it too XD
> 3. Yeah, I see the problem, it is the behavior from the original annotation box, apparently it has some min height set, I will see if I can adjust it.
> 4. Yes, we also asked Piotr about this and he said the title technically should be annotated too. But I think the current code should already work for that (?) -- in your imported file, do you have the text with the type "title"?

2. aha XD okie so we can ask them in the next meeting maybe XD
3. okie XD if it cant be I think right now is oke, it's good enough XD
4. ah sorry, it's my mistake XD now it's oke

haanh764 commented:
> Because if the title is only in metadata, we cant annotate it sweat_smile we need to have an element in the array in the imported file, which contains the text and the type: "title"

Yeah XD sorry, I used wrong file haha XD

teddy-f-47 commented:
> 2. aha XD okie so we can ask them in the next meeting maybe XD
> 3. okie XD if it cant be I think right now is oke, it's good enough XD
> 4. ah sorry, it's my mistake XD now it's oke

1. I have fixed it. 😅 Now the dialog won't appear when the checkbox is marked, so the user shouldn't be able to add texts when the textfield should be disabled.
2. I have added the min and max values to the sliders, so they now looks something like this:
<img width="316" alt="ss2" src="https://user-images.githubusercontent.com/64476430/194633660-c4ac1622-3e48-45f3-8cf5-0af437268b2f.PNG">
3. Unfortunately, I am not able to eliminate the whitespace under the annotation box XD I think something in the `entity-editor` component keeps automatically adding extra whitespace at the bottom, I don't know why. However, I have increase the input component's height a bit:
<img width="960" alt="ss3" src="https://user-images.githubusercontent.com/64476430/194633878-dce2f78a-85b4-428e-93e8-4365d8c93509.PNG">

artikandri reviewed (approved):
I have tested it and it works well over here, good job! I will give my approval for now, but let's wait for Piotr and Mr Jan's feedbacks before merging it XD

haanh764 commented:
I have tested it again and it works well now XD Thank you for your great work, approved XD

teddy-f-47 commented:
Hi all, so I have adjusted the css a tiny bit to prevent some longer words in English and French from breaking awkwardly XD it now looks like this:
<img width="316" alt="ss1" src="https://user-images.githubusercontent.com/64476430/194857157-180922da-f39b-48dc-8266-247d0320ea2a.PNG">

I think in Polish and the other languages it looks quite fine. I will merge this PR for now. 😅
